vm/src/stdlib/ctypes/array.rs

```diff
@@ -0,0 +1,633 @@
+                    value.class().name
+                )))
+            }
+        } else {
+            // unicode string zelf._type_ == "u"
+            if let Ok(value) = value.clone().downcast_exact::<PyStr>(vm) {
+                let wide_str =
+                    unsafe { WideCString::from_str_with_nul_unchecked(value.to_string()) };
+
+                let wide_str_len = wide_str.len();
+
+                if wide_str.len() > my_size {
+                    Err(vm.new_value_error("string too long".to_string()))
+                } else {
+                    let res = if cfg!(windows) {
+                        wide_str
+                            .into_vec()
+                            .iter_mut()
+                            .map(|i| u16::to_ne_bytes(*i as u16).to_vec())
+                            .flatten()
+                            .collect::<Vec<u8>>()
+                    } else {
+                        wide_str
+                            .into_vec()
+                            .iter_mut()
+                            .map(|i| u32::to_ne_bytes(*i).to_vec())
+                            .flatten()
+                            .collect::<Vec<u8>>()
+                    };
+
+                    bytes[0..wide_str_len].copy_from_slice(res.as_slice());
+
+                    Ok(())
+                }
+            } else {
+                Err(vm.new_value_error(format!(
+                    "unicode string expected instead of {} instance",
+                    value.class().name
+                )))
+            }
+        }
+    }
+
+    #[pyproperty(name = "raw")]
+    fn raw(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyResult<PyObjectRef> {
+        // zelf._type_ == "c"
+
+        let obj = zelf.as_object();
+        let buffer = try_buffer_from_object(vm, obj)?;
+        let buffer_vec = buffer.obj_bytes().to_vec();
+
+        Ok(PyBytes::from(buffer_vec).into_object(vm))
+    }
+
+    #[pyproperty(name = "raw", setter)]
+    fn set_raw(zelf: PyRef<Self>, value: PyObjectRef, vm: &VirtualMachine) -> PyResult<()> {
+        let obj = zelf.as_object();
+        let my_buffer = try_buffer_from_object(vm, obj)?;
+        let my_size = my_buffer.get_options().len;
+
+        let new_value = try_buffer_from_object(vm, &value)?;
+        let new_size = new_value.get_options().len;
+
+        // byte string zelf._type_ == "c"
+        if new_size > my_size {
+            Err(vm.new_value_error("byte string too long".to_string()))
+        } else {
+            let mut borrowed_buffer = my_buffer.obj_bytes_mut();
+            let src = new_value.obj_bytes();
+            borrowed_buffer[0..new_size].copy_from_slice(&src);
+            Ok(())
+        }
+    }
+
+    #[pymethod(name = "__len__")]
+    fn len(&self) -> usize {
+        self._length_
+    }
+
+    #[pymethod(magic)]
+    fn getitem(
+        zelf: PyRef<Self>,
+        k_or_idx: Either<isize, PyRef<PySlice>>,
```

Comment:
`rustpython_vm::sliceable::SequenceInex` type fits here

Reply:
Hm, that's new right? I couldn't find it in my local repo.

Reply:
Like this, right (inside `sliceable.rs`)?
```rust
fn get_item(
    &self,
    vm: &VirtualMachine,
    needle: PyObjectRef,
    owner_type: &'static str,
) -> PyResult<Either<Self::Item, Self::Sliced>> {
    let needle = SequenceIndex::try_from_object_for(vm, needle, owner_type)?;
    match needle {
        SequenceIndex::Int(value) => {
            let pos_index = self.wrap_index(value).ok_or_else(|| {
                vm.new_index_error(format!("{} index out of range", owner_type))
            })?;
            Ok(Either::A(self.do_get(pos_index)))
        }
        SequenceIndex::Slice(slice) => Ok(Either::B(self.get_slice_items(vm, &slice)?)),
    }
}
```

Reply:
Yes, exactly